pytradfri/api/aiocoap_api.py

```diff
@@ -30,12 +32,16 @@ def _get_psk(self, host, port):
 
 
 class APIFactory:
+    last_changed = time.time()
+
     def __init__(self, host, psk_id='pytradfri', psk=None, loop=None):
         self._psk = psk
         self._host = host
         self._psk_id = psk_id
         self._loop = loop
-        self._observations_err_callbacks = []
+        self._is_checking = False
```

Comment:
Are you expecting a race of some sort? Maybe we should consider using `asyncio.Lock` instead?

Reply:
Not specifically a race condition. But because the `asyncio.sleep` function is called within `_remove_timedout_observations`, it can happen that a observation is checked multiple times. This shouldn't cause any problems, but is just unnecessary. Using an `asyncio.Lock` will have the same behaviour, but just seqential instead of parallel.

This check is only done to make sure we check and reset the observations only once at a time, even when controlling multiple lights at once.